src/control/xml/XmlPointNode.cpp

```diff
@@ -23,13 +14,12 @@ void XmlPointNode::writeOut(OutputStream* out) {
 
     out->write(">");
 
-    for (GList* l = this->points; l != nullptr; l = l->next) {
-        auto* p = static_cast<Point*>(l->data);
-        if (l != this->points) {
+    for (auto pointIter = points.begin(); pointIter != points.end(); pointIter++) {
+        if (pointIter != points.begin()) {
```

Comment:
I would move the condition out of the loop, also prepend the ++ (increment) in c++, because val++ will create a copy of val before increment. And not every compiler is capable to remove the extra code.